### ISSUE TITLE
Improve quote handling in command-line

### ISSUE THREAD

**Issue by dleehr** [Duke-GCB/calrissian#37]
In attempts to run [exomeseq-gatk4-preprocessing](https://github.com/Duke-GCB/bespin-cwl/blob/master/workflows/exomeseq-gatk4-preprocessing.cwl), the map step fails because the command is not constructed correctly.

Relevant snippet from generated spec:

```
  containers:
    - args:
        - >-
          bash bwa-mem-samtools.sh -R
          @RG\tID:SA05051\tLB:F012354\tPL:Illumina\tPU:SA05051\tSM:SA05051 -t 20
          /var/lib/cwl/stgc9814adc-1982-403b-ade3-e106d7a82960/human_g1k_v37_decoy.fasta
          /var/lib/cwl/stgec2e16b0-4425-4dcc-8baa-dd9b5bef0451/SA05051-R1_val_1.fq.gz
          /var/lib/cwl/stg4f3a9d6b-7759-40fb-8070-ffecf1403d45/SA05051-R2_val_2.fq.gz
          > SA05051-mapped.bam
      command:
        - /bin/sh
        - '-c'
      env:
        - name: HOME
          value: /plpTRq
        - name: TMPDIR
          value: /tmp
      image: 'broadinstitute/genomes-in-the-cloud:2.3.1-1512499786'
```

I believe that the base cwltool job would catch this with 

https://github.com/common-workflow-language/cwltool/blob/bbe20f54deea92d9c9cd38cb1f23c4423133d3de/cwltool/job.py#L243

but this feature is still outstanding in #8

**Comment by dleehr** [comment]
The args here definitely need shell quoting - both the `@` and `\t` characters match the [needs_shell_quoting_re](https://github.com/common-workflow-language/cwltool/blob/bbe20f54deea92d9c9cd38cb1f23c4423133d3de/cwltool/job.py#L44):

```
>>> import re
>>> needs_shell_quoting_re =  re.compile(r"""(^$|[\s|&;()<>\'"$@])""")
>>> arg = "@RG\tID:SA05051\tLB:F012354\tPL:Illumina\tPU:SA05051\tSM:SA05051"
>>> needs_shell_quoting_re.findall(arg)
['@', '\t', '\t', '\t', '\t', '\t']
```

**Comment by dleehr** [comment]
Upon further inspection it looks like cwltool only uses the `needs_shell_quoting_re` when logging, so it's not part of the actual command.

But I believe it's still needed as it's passed as a shell script instead of a command

See Reana


https://github.com/reanahub/reana-workflow-engine-cwl/blob/57ff1d8b315d583017e92a868a7f409cd018a21d/reana_workflow_engine_cwl/cwl_reana.py#L186-L188

**Comment by dleehr** [comment]
This was implemented by #39
